crates/vm/levm/src/opcode_handlers/environment.rs

```diff
@@ -185,7 +192,14 @@ impl VM {
             let mut padded_calldata = vec![0u8; size];
             let data_len_to_copy = calldata.len().min(size);
 
-            padded_calldata[..data_len_to_copy].copy_from_slice(&calldata[..data_len_to_copy]);
+            padded_calldata
```

Comment:
We could probably refactor this to avoid handling this.

Reply:
Done!

Reply:
https://github.com/lambdaclass/lambda_ethereum_rust/pull/1107/commits/e9b8381909a60a40c7ed64a95189e0cd8665b563